frontend/store/wallet.ts

```diff
@@ -295,9 +298,12 @@ export const actions = {
     async setup({ commit, dispatch, getters }: ActionContext<State, State>): Promise<void> {
         commit('reset');
         if (!getters.isConnected) {
-            await dispatch('autoConnect');
+            if (!WAS_WALLET_DISCONNECTED) {
+                await dispatch('autoConnect');
+            }
         }
         dispatch('refetch');
         dispatch('surplus/setup', undefined, { root: true });
+        WAS_WALLET_DISCONNECTED = false;
```

Comment:
Using a local variable is a very hacky approach, since calling `setup` twice will no longer work and will autoconnect the wallet again. It just doesn't fix the problem, but introduces more side effects.

Instead, I would remove `await dispatch('autoConnect')` from the setup action (since it's only need to be called on the page load and not with every `setup`) and move it to the place where we call `setup` at the page load. I guess right before this line: https://github.com/sidestream-tech/unified-auctions-ui/blob/19a386750245af34e194f8717cc0d9495aa9db8d/frontend/plugins/vuex-persist.client.js#L17